_posts/2020-04-23-tensorflow-real-time.md

```diff
@@ -0,0 +1,22 @@
+---
+layout: post
+title: Real-time deep learning predictions with Essentia
+category: news
+---
+Due to the temporal nature of sound, real-time constraints are a requirement for many related signal processing applications. Examples of this are the modern encoding algorithms that compress the audio and video streams preventing teleconferences from all around the world from collapsing the internet or the modern mixing consoles allowing the simultaneous use of hundreds of audio effects to improve the quality of live performances.
+
+As a continuation of this idea, modern approaches relying on deep learning must be able to operate under the same constraints to be usable in many deployment scenarios. To this end, we have equipped Essentia with algorithms that support real-time deep learning inference.
+
+The real-time capability, however, relies ultimately on the complexity of the model and the computational resources available. In this sense, some of our models are compact enough to perform real-time predictions with the CPU of a regular laptop and we have created an open demo exemplifying this!
+
+Our demo relies on [SoundCard](https://soundcard.readthedocs.io/en/latest/) to catch the computer audio loopback (whatever is being played from Youtube, Spotify, your local player...) and stream it into Essentia. We're using our auto-tagging model based in the MusiCNN architecture, already introduced on a [previous post](https://mtg.github.io/essentia-labs/news/2019/10/19/tensorflow-models-in-essentia/). We have adapted our model to operate on chunks of one second of audio. This process does not require any extra training, but it reduces the performance of the model a bit. The video below shows a moving window with the mel bands used for prediction on the left and the activated tags on the right.
```

Comment:
"We use a pre-trained auto-tagging model based on the MusiCNN architecture introduced in our [previous post](https://mtg.github.io/essentia-labs/news/2019/10/19/tensorflow-models-in-essentia/), that we adapted to operate on shorter one-second chunks of audio. This process did not require any additional training, but it reduced the performance of the model a bit. For predictions, we compute a Mel-spectrogram of each audio chunk as an input to this model. 

The video below shows a moving window with the Mel-spectrograms used for prediction on the left and the resulting tag activations on the right."